File: type-utils/src/main/java/gov/nasa/jpl/aerie/types/DirectiveActivitySource.java
package gov.nasa.jpl.aerie.types;

import org.apache.commons.lang3.tuple.Pair;

public final class DirectiveActivitySource implements ActivitySource<Pair<ActivityDirective, Long>> {
  Pair<ActivityDirective, Long> value;

  public DirectiveActivitySource(ActivityDirective directive, Long directiveId) {
    value = Pair.of(directive, directiveId);
  }

  @Override
  public Pair<ActivityDirective, Long> getValue() {
    return value;
  }
}

File: procedural/timeline/src/main/kotlin/gov/nasa/ammos/aerie/procedural/timeline/payloads/reference/DirectiveActivitySource.kt
package gov.nasa.ammos.aerie.procedural.timeline.payloads.reference

import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.AnyDirective
import gov.nasa.ammos.aerie.procedural.timeline.payloads.activities.Directive

data class DirectiveActivitySource (
  override val v: Pair<Directive<AnyDirective>, Long>
): ActivitySource<Pair<Directive<AnyDirective>, Long>>(v)
